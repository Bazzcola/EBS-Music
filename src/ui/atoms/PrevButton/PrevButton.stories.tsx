import React from 'react';
import { PrevButton } from 'ui/atoms/PrevButton/PrevButton';

export default {
  title: 'Atoms/PrevButton',
  component: PrevButton
};
export const Button4 = () => {
  return <PrevButton />;
};
