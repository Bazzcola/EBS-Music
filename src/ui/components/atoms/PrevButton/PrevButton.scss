.prev_button {
    font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
    font-weight: 700;
    border: 0;
    border-radius: 50%;
    cursor: pointer;
    display: inline-block;
    color: white;
    min-width: 60px;
    min-height: 60px;
    text-align: center;
    background: none;
}

.prev_button:hover {
    background: rgb(224, 224, 224);
}

.svg_prevbtn {
    display: flex;
    align-items: center;
    width: 20px;
    height: 28px;
    color: black;
    fill: currentColor;
    stroke: currentColor;
    stroke-width: 0;
    vertical-align: top;
    margin: 0 auto;
}