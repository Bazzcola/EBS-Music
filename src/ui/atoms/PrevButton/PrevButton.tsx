import React, { useState } from 'react';
import 'ui/atoms/PrevButton/PrevButton.scss';

export const PrevButton = () => {
  const [active, setActive] = useState<boolean>(false);
  const changes = () => {
    setActive(!active);
  };
  return (
    <button onClick={changes} className="prev_button">
      <svg className="svg_prevbtn" viewBox="0 0 12 12">
        <path d="M2 5.397V.5a.5.5 0 1 0-1 0v11a.5.5 0 1 0 1 0V6.597L11 12V0L2 5.397z"></path>
      </svg>
    </button>
  );
};
