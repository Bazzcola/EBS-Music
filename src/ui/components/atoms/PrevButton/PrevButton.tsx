import React, { useState, useContext, useEffect, useCallback } from 'react';
import { AudioContext, song } from 'ui/context/audioContext';
import { Tracks } from 'ui/context/audioContext';
import './PrevButton.scss';

export const PrevButton = () => {
  const [files, setFiles] = useState<string[]>(['']);
  const {
    counter,
    setCounter,
    setPlaying,
    setAudioFiles,
    profile
  } = useContext(AudioContext);

  useEffect(() => {
    setFiles(Tracks.map((item: song) => item.src));
  }, []);

  let keys: any = Object.keys(files);

  const prevSong = useCallback(
    (e: { preventDefault: () => void }) => {
      e.preventDefault();
      if (profile === false) {
        if (counter === 0) {
          setCounter(files.length - 1);
          setAudioFiles(files[keys[files.length - 1]]);
        } else {
          setCounter((count: number) => count - 1);
          setAudioFiles(files[keys[counter - 1]]);
          setPlaying(true);
        }
      } else {
        return;
      }
    },
    [counter, files, keys, profile, setAudioFiles, setCounter, setPlaying]
  );

  return (
    <button
      onClick={prevSong}
      className={`prev_button ${profile ? 'prev_button_active' : ''}`}
    >
      <svg className="svg_prevbtn" viewBox="0 0 12 12">
        <path d="M2 5.397V.5a.5.5 0 1 0-1 0v11a.5.5 0 1 0 1 0V6.597L11 12V0L2 5.397z"></path>
      </svg>
    </button>
  );
};
