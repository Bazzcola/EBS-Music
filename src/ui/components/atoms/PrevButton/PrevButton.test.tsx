import React from 'react';
import { render } from '@testing-library/react';
import { PrevButton } from './PrevButton';

it('render PrevButton correct', () => {
  const button = () => {
    render(<PrevButton />);
  };
  expect(button).toMatchSnapshot();
});
