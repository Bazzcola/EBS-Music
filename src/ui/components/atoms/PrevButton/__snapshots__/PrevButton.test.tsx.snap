// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PrevButton correct 1`] = `[Function]`;
